prototype/FFT_image_processing/arduino_image_processing/arduino_image_processing.ino

```diff
@@ -0,0 +1,110 @@
+
+#define fft_real float
+
+#include "FFT.h"
+
+void setup() {
+  // initialize serial:
+  Serial.begin(115200, SERIAL_8N1);
+  pinMode(LED_BUILTIN, OUTPUT);
+  digitalWrite(LED_BUILTIN, LOW);
+}
+
+unsigned FFTLength = 0;
+int dir = 0;
+int fftCode = 0;
+
+void loop() {
+  if (Serial.available() && fftCode == 0) {
+
+    digitalWrite(LED_BUILTIN, HIGH);
+
+    // FFT over serial
+
+    // Reading
+    // read length
+    readElement(&FFTLength, sizeof(FFTLength));
+
+    // init arrays
+    fft_real FFTreals[FFTLength];
+    fft_real FFTimgs[FFTLength];
+    memset(FFTreals, 0, FFTLength * sizeof(fft_real));
+    memset(FFTimgs, 0, FFTLength * sizeof(fft_real));
+
+    // read reals
+    readArray(FFTLength, FFTreals, sizeof(fft_real));
+    // read imaginaries
+    readArray(FFTLength, FFTimgs, sizeof(fft_real));
+    // read direction
+    readElement(&dir, sizeof(dir));
+
+    // Execute FFT
+    fftCode = FFT(FFTLength, FFTreals, FFTimgs, dir);
+    if (fftCode != 0) {
+      digitalWrite(LED_BUILTIN, LOW);
+      return;
+    }
+
+    // Writing
+    // write reals
+    writeArray(FFTLength, FFTreals, sizeof(fft_real));
+    // write imaginaries
+    writeArray(FFTLength, FFTimgs, sizeof(fft_real));
+
+  } else {
+    digitalWrite(LED_BUILTIN, LOW);
+  }
+}
+
+/**
+ * @brief Read an array of length arraySize, where each element is of size
+ * sizeOfElement. This function will block untill all elements are read.
+ * @param arraySize The size of the array to read.
+ * @param outArray The array in which to store the result of the read data.
+ * @param sizeOfElement The size of each element in the array.
+ */
+int readArray(const uint32_t arraySize, void* outArray,
```

Comment:
I believe it would be very interesting to put theses methods in a seperate .ino because everyone will need them.

Reply:
Only one `.ino` file may be used for an Arduino sketch.
Therefore the files `arduino_serial_port.cpp` and `arduino_serial_port.h` were created for this purpose.